src/service/ConfigLoader.ts

```diff
@@ -124,8 +127,12 @@ export class ConfigLoader {
 
         if (presetData.nemesis) {
             if (oldAddresses) {
+                if (!oldPresetData) {
+                    throw new Error('oldPresetData must be defined when upgrading!');
+                }
                 // Nemesis configuration cannot be changed on upgrade.
                 addresses.mosaics = oldAddresses.mosaics;
+                presetData.nemesis = oldPresetData.nemesis;
```

Comment:
should you check that nemesis is the same ?

Reply:
I've removed that line. It's not currently required in this PR. 

I'm planning to provide a new PR improving how nemesis block is generated.
- Allow users to provide all the accounts, so the same custom preset executed twice (from scratch) generates the same accounts, balances, and nemesis block. 
- Allow users to include balances for both main and harvesting accounts (atm only the main mosaic can be 'opted-in')
- Validations that node's main accounts are not over the harvesting limit when dynamically generated (no user-provided)